src/Persister/ElasticSearchPersister.php

```diff
@@ -87,22 +133,95 @@ public function reuseTransactionId($use = true)
         $this->useTransactionId = $use;
     }
 
+    /**
+     * Sets the client connection to elastic search.
+     *
+     * @param Elastica\Client $connection The conneciton to elastic search
+     * @return $this
+     */
+    public function setConnection(Client $connection)
+    {
+        $this->connection = $connection;
+
+        return $this;
+    }
+
+    /**
+     * Gets the client connection to elastic search.
+     *
+     * If connection is not defined, create a new one.
+     *
+     * @return Elastica\Client
+     */
+    public function getConnection()
+    {
+        if ($this->connection === null) {
+            $this->connection = ConnectionManager::get('auditlog_elastic');
+        }
+
+        return $this->connection;
+    }
+
     /**
      * Sets the client connection to elastic search when passed.
      * If no arguments are provided, it returns the current connection.
      *
+     * @deprecated Use getConnection()/setConnection() instead
      * @param Elastica\Client $connection The conneciton to elastic search
      * @return Elastica\Client
      */
     public function connection(Client $connection = null)
     {
-        if ($connection === null) {
-            if ($this->connection === null) {
-                $this->connection = ConnectionManager::get('auditlog_elastic');
-            }
-            return $this->connection;
+        if ($connection !== null) {
+            return $this->setConnection($connection);
         }
 
-        return $this->connection = $connection;
+        return $this->getConnection();
+    }
+
+    /**
+     * Sets the Elasticsearch index used to store events
+     *
+     * @param string $index Name of the Elasticsearch index
+     * @return $this
+     */
+    public function setIndex($index)
+    {
+        $this->index = $index;
+
+        return $this;
+    }
+
+    /**
+     * Gets the Elasticsearch index to persist events
+     *
+     * @return string Name of the Elasticsearch index
+     */
+    public function getIndex()
+    {
+        return sprintf($this->index, '-' . gmdate('Y.m.d'));
```

Comment:
I don't like this is forcing the index to always have a date. This was optional before.

Also, considering that the index name is now the type, I think the new format should be `gmdate('Y.m.d')-$this->index` instead. So what you can find all events for a year, month or day using a prefix.

Reply:
Is still optional. As before, `sprintf` will replace only `%s` part of  `$this->index`, if is there. Otherwise, as before, you will get `$this->index` without replacement.

Reply:
My bad, I totally missed the `sprintf` call, sorry. What about my suggestion of inverting the order?

Reply:
Make sense for me (invert the order).

But I don't think is possible: https://www.elastic.co/guide/en/elasticsearch/reference/current/date-math-index-names.html

Reply:
ah cool, if it is already a feature in Elastic then no need to worry. In Elastic 2 you had to do something like `my_index-2018.*`